test/mpi/runtests

```diff
@@ -933,169 +935,177 @@ sub get_resultTest {
     }
 }
 
-sub TestNormal {
-    my ($MPIOUT, $programname) = @_;
-    my $found_error = 0;
+sub check_result {
+    my $MPIOUT = shift;
+
     my $found_noerror = 0;
+    my $test_skipped = 0;
+    my $run_status = 0;
+    my $stray_output = 0;
+    my $warning_output = 0;
     my $inline = "";
     while (<$MPIOUT>) {
         print STDOUT $_ if $g_opt{verbose};
-        # Skip FORTRAN STOP
-        if (/FORTRAN STOP/) { next; }
         $inline .= $_;
+
         if (/^\s*No [Ee]rrors\s*$/ && $found_noerror == 0) {
             $found_noerror = 1;
         }
-        elsif (/^srun: error: .*: signal: Communication connection failure/) {
+        elsif (/^\s*Test Skipped\s*$/) {
+            $test_skipped++;
+        }
+        elsif (/^WARNING: /) {
+            $warning_output++;
+        }
+        elsif (/(FORTRAN STOP|requesting checkpoint|checkpoint completed)\s*$/) {
             # skip
         }
-        elsif (!/^\s*Test (Passed|Skipped)\s*$/ && !/requesting checkpoint\s*$/ && !/checkpoint completed\s*$/) {
-            print STDERR "Unexpected output in $programname: $_";
-            $found_error = 1;
+        elsif (/^srun: error: .*: signal: Communication connection failure/) {
+            # skip
         }
-    }
-    if ($found_noerror == 0) {
-        if ($inline !~ /Test Skipped/) {
-            print STDERR "Program $programname exited without No Errors\n";
-            $found_error = 1;
+        else {
+            $stray_output++;
         }
     }
     my $rc = close ($MPIOUT);
-    if ($rc == 0) {
-        # Only generate a message if we think that the program
-        # passed the test.
-        if (!$found_error) {
-            my $run_status = $?;
-            my $signal_num = $run_status & 127;
-            if ($run_status > 255) { $run_status >>= 8; }
-            print STDERR "Program $programname exited with non-zero status $run_status\n";
-            if ($signal_num != 0) {
-                print STDERR "Program $programname exited with signal $signal_num\n";
-            }
-            $found_error = 1;
-        }
+    if (!$rc) {
+        $run_status = $?;
     }
-    return ($found_error, $inline);
+
+    return ($found_noerror, $test_skipped, $run_status, $stray_output, $warning_output, $inline);
 }
-# ----------------------------------------------------------------------------
-#
-# TestStatus is a special test that reports success *only* when the 
-# status return is NONZERO
-sub TestStatus {
-    my $MPIOUT = $_[0];
-    my $programname = $_[1];
-    my $found_error = 0;
 
-    my $inline = "";
-    while (<$MPIOUT>) {
-        #print STDOUT $_ if $g_opt{verbose};
-        # Skip FORTRAN STOP
-        if (/FORTRAN STOP/) { next; }
-        $inline .= $_;
-        # ANY output is an error. We have the following output
-        # exception for the Hydra process manager.
-        if (/=*/) { last; }
```

Comment:
This exception is crazy. It exempts for `errors/comm/subcomm_abort` for the message
```
application called MPI_Abort(comm=0x84000004, 8) - process 0
```